src/sentry/taskworker/registry.py

```diff
@@ -79,9 +81,61 @@ def wrapped(func: Callable[P, R]) -> Task[P, R]:
     def send_task(self, activation: TaskActivation) -> None:
         # TODO(taskworker) producer callback handling
         self.producer.produce(
-            ArroyoTopic(name=self.topic),
+            ArroyoTopic(name=self.topic.value),
             KafkaPayload(key=None, value=activation.SerializeToString(), headers=[]),
         )
 
 
-# TODO(taskworker) Import TaskRegistrys
+class TaskRegistry:
+    """
+    Registry of all namespaces.
+
+    The TaskRegistry is responsible for handling namespace -> topic resolution
+    during startup.
+    """
+
+    def __init__(self) -> None:
+        self._namespaces: dict[str, TaskNamespace] = {}
+        self._router = self._build_router()
+
+    def _build_router(self) -> TaskRouter:
+        router_name: str = settings.TASKWORKER_ROUTER
+        router_class = import_string(router_name)
+        router = router_class()
+        assert hasattr(router, "route_namespace")
+
+        return router
+
+    def get(self, name: str) -> TaskNamespace:
+        """Fetch a namespace by name."""
+        if name not in self._namespaces:
+            raise KeyError(f"No task namespace with the name {name}")
+        return self._namespaces[name]
+
+    def get_task(self, namespace: str, task: str) -> Task[Any, Any]:
+        """Fetch a task by namespace and name."""
+        return self.get(namespace).get(task)
+
+    def import_tasks(self) -> None:
+        """Import all the modules listed in settings.TASKWORKER_IMPORTS"""
+        imports = settings.TASKWORKER_IMPORTS
+        for module in imports:
+            __import__(module)
+
+    def create_namespace(self, name: str, retry: Retry | None = None) -> TaskNamespace:
```

Comment:
Where will this be used in the future?

Reply:
This would be used by application modules as part of the application startup. For example `sentry.integrations.tasks` could have something like

```python
# In src/sentry/integrations/tasks/__init__.py
from sentry.taskworker.registry import taskregistry

integration_tasks = taskregistry.create_namespace("integrations")

# in src/sentry/integrations/tasks/do_things.py
from sentry.integrations.tasks import integration_tasks

@integration_tasks.register(name="sentry.integrations.do_thing")
def do_thing(org_id: int) -> None:
    ... 
```